src/main/kotlin/ecommerce/controller/ProductController.kt

```diff
@@ -0,0 +1,58 @@
+package ecommerce.controller
+
+import ecommerce.model.Product
+import ecommerce.service.ProductService
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.DeleteMapping
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.PutMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RestController
+import java.net.URI
+
+@RestController
+class ProductController(private val productService: ProductService) {
+    @PostMapping("/products")
+    fun createProduct(
+        @RequestBody newProduct: Product,
+    ): ResponseEntity<Product> {
```

Comment:
Is it okay to use a model object as a request or a response?
If so, why do you think that’s a good idea? If not, what’s the best practice?